lua/neotest-go/init.lua

```diff
@@ -220,7 +220,10 @@ function adapter.prepare_results(tree, lines, go_root, go_module)
           results[value.id].errors = errors
         end
         if test_result.status == test_statuses.fail then
-          results[file_id].status = test_statuses.fail
+          if file_id ~= nil then
+            -- when executing a function there may be no file_id
+            results[file_id].status = test_statuses.fail
+          end
```

Comment:
I think `file_id` is always `nil` here? So I think that the if statement can be removed completely.

Reply:
I've seen this value to be `nil` when executing a `t.Run` test function.